src/createServer.js

```diff
@@ -1,8 +1,46 @@
 'use strict';
 
+const http = require('http');
+const path = require('path');
+const fsPromises = require('fs').promises;
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  return http.createServer(async (req, res) => {
+    res.setHeader('Content-Type', 'text/plain');
+
+    const url = new URL(req.url, `http://${req.headers.host}`);
+
+    if (!url.pathname.startsWith('/file')) {
+      res.statusCode = 400;
+      res.statusMessage = 'Path should start with "/file/';
+      res.end();
+
+      return;
+    }
+
+    const requestedPath = url.pathname.slice(6) || 'index.html';
```

Comment:
The logic for extracting the requested path assumes that the path always starts with '/file/'. Ensure that the path slicing is correct and consider edge cases where the path might be shorter than expected.